lib/web/mfa.go

```diff
@@ -211,3 +212,155 @@ func getDeviceUsage(reqUsage string) (proto.DeviceUsage, error) {
 
 	return deviceUsage, nil
 }
+
+type isMFARequiredDatabase struct {
+	// ServiceName is the database service name.
+	ServiceName string `json:"serviceName"`
+	// Protocol is the type of the database protocol.
+	Protocol string `json:"protocol"`
+	// Username is an optional database username.
+	Username string `json:"username,omitempty"`
+	// Name is an optional database name.
+	Name string `json:"name,omitempty"`
+}
+
+type isMFARequiredKube struct {
+	// Name is the name of the kube cluster.
+	Name string `json:"name"`
+}
+
+type isMFARequiredNode struct {
+	// Name can be node's hostname or UUID.
+	Name string `json:"name"`
+	// Login is the OS login name.
+	Login string `json:"login"`
+}
+
+type isMFARequiredWindowsDesktop struct {
+	// Name is the Windows Desktop server name.
+	Name string `json:"name"`
+	// Login is the Windows desktop user login.
+	Login string `json:"login"`
+}
+
+type isMfaRequiredRequest struct {
+	// Database contains fields required to check if target database
+	// requires MFA check.
+	Database *isMFARequiredDatabase `json:"database,omitempty"`
+	// Node contains fields required to check if target node
+	// requires MFA check.
+	Node *isMFARequiredNode `json:"node,omitempty"`
+	// WindowsDesktop contains fields required to check if target
+	// windows desktop requires MFA check.
+	WindowsDesktop *isMFARequiredWindowsDesktop `json:"windowsDesktop,omitempty"`
+	// KubeCluster is the name of the kube cluster to check if target cluster
+	// requires MFA check.
+	KubeCluster *isMFARequiredKube `json:"kube,omitempty"`
+}
+
+func (r *isMfaRequiredRequest) checkAndGetProtoRequest() (*proto.IsMFARequiredRequest, error) {
+	numRequests := 0
+	var protoReq *proto.IsMFARequiredRequest
+
+	if r.Database != nil {
+		numRequests++
+		if r.Database.ServiceName == "" {
+			return nil, trace.BadParameter("missing name for checking database target")
+		}
+		if r.Database.Protocol == "" {
+			return nil, trace.BadParameter("missing protocol for checking database target")
+		}
+
+		protoReq = &proto.IsMFARequiredRequest{
+			Target: &proto.IsMFARequiredRequest_Database{Database: &proto.RouteToDatabase{
+				ServiceName: r.Database.ServiceName,
+				Protocol:    r.Database.Protocol,
+				Database:    r.Database.Name,
+				Username:    r.Database.Username,
+			}},
```

Comment:
nit:

```suggestion
			Target: &proto.IsMFARequiredRequest_Database{
				Database: &proto.RouteToDatabase{
					ServiceName: r.Database.ServiceName,
					Protocol:    r.Database.Protocol,
					Database:    r.Database.Name,
					Username:    r.Database.Username,
				},
			},
```

Similarly for others.